software/fishstick/src/joybus/joybus.rs

```diff
@@ -1,4 +1,12 @@
+use bitvec::prelude::*;
+use bitvec::BitArr;
+
+pub const JOYBUS_SIZE: usize = 128;
+
+pub type Packet = BitArr!(for JOYBUS_SIZE, in u8, Msb0);
+
 pub trait Joybus {
-    fn send(&self, packet: &[i32]) -> Result<(), ()>;
-    fn send_receive<const S: usize>(&self, packet: &[i32]) -> Result<[i32; S], ()>;
+    fn send(&self, packet: &Packet) -> Result<(), ()>;
+    fn send_receive(&self, packet: &Packet) -> Result<&Packet, ()>;
+    fn send_receive_no_alloc(&self, packet: &Packet, receive_packet: &mut Packet) -> Result<(), ()>;
```

Comment:
For now, we're just assuming input packet length == output packet length, right?

Reply:
yes this is what I am thinking for now. we will ultimately need to change the packet length later as well.

also at some point we could switch back to generics and remove the constant altogether